jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java

```diff
@@ -869,12 +869,44 @@ public List<String> getSoftDeleteProperties(Class<?> clazz) {
      * @return true if entity's {@link EntityEntry} implements {@link EntityEntryHasUuid}
      */
     public boolean hasUuid(MetaClass metaClass) {
-        return metaClass.getAnnotations().containsKey(UUID_KEY_ANN_NAME);
+        String uuidPropertyName = getUuidPropertyName(metaClass);
+        return uuidPropertyName != null;
     }
 
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param clazz entity java class
+     * @return uuid property name or null if it doesn't exist
+     */
     @Nullable
     public String getUuidPropertyName(Class<?> clazz) {
-        return (String) metadata.getClass(clazz).getAnnotations().get(UUID_KEY_ANN_NAME);
+        MetaClass metaClass = metadata.getClass(clazz);
+        return getUuidPropertyName(metaClass);
+    }
+
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param metaClass entity MetaClass
+     * @return uuid property name or null if it doesn't exist
+     */
+    @Nullable
+    public String getUuidPropertyName(MetaClass metaClass) {
```

Comment:
Do we really need to introduce a new `getUuidPropertyName` method? Why not to implement the behavior in the existing `getUuidPropertyName(Class<?> clazz)`?

Reply:
To iliminate double conversion `MetaClass` -> `JavaClass` -> `MetaClass`:
`hasUuid` accepts `MetaClass` and uses result of `getUuidPropertyName` - original one accepts `Class<?>` and convert it to `MetaClass`. Not a critical point, we can keep this conversion and remove extra method.